code/game/atoms.dm

```diff
@@ -958,3 +958,56 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 		else if(C)
 			color = C
 			return
+
+
+/** Call this when you want to present a renaming prompt to the user.
+
+    It's a simple proc, but handles annoying edge cases such as forgetting to add a "cancel" button,
+	or being able to rename stuff remotely.
+
+	Arguments:
+	* user - the renamer.
+	* implement - the tool doing the renaming (usually, a pen).
+	* freeform - whether we allow to change the name arbitrarily, or if it should go "thing - user-given label".
+	* actually_rename - whether we want to really change the `src.name`, or if we want to do everything *except* that.
+	* prompt - a custom "what do you want rename this thing to be?" prompt shown in the inpit box.
+
+	Returns: Either null if the renaming was aborted, or the user-provided sanitized string.
+ **/
+/atom/proc/rename_interactive(mob/user, obj/implement = null, freeform = FALSE,
+		actually_rename = TRUE, prompt = null)
+	// Sanity check that the user can, indeed, rename the thing.
+	if((implement && implement.loc != user) || !in_range(src, user))
```

Comment:
I could be checking for user being alive, conscious, literate, etc.
I could also be checking that the pen is in active hand (rather than anywhere on the person), as some of the previous code was doing.

I don't do the former because I'm lazy, and the latter because i would *want* to break "realism" for a bit easier gameplay.

Reply:
Trying to rewrite things using Telekinesis will now cause runtimes. Most likely not what you'd want to happen